File: internal/front/src/schemas/book.ts
import { z } from 'zod'

import { camelize } from '@/schemas/common'

export const Book = z
  .object({
    title: z.string(),
    isbn13: z.string().optional(),
    isbn10: z.string().optional(),
    publication_year: z.number(),
    price: z.number(),
    image_url: z.string().optional(),
    edition: z.string().optional(),
    publisher: z.string(),
    authors: z.string().array(),
  })
  .transform(camelize)

File: internal/models/book.go
package models

import "github.com/atsuyaourt/xyz-books/internal/util"

type Book struct {
	Title           string   `json:"title"`
	ISBN13          string   `json:"isbn13"`
	ISBN10          string   `json:"isbn10"`
	Price           float64  `json:"price"`
	PublicationYear int64    `json:"publication_year"`
	ImageUrl        string   `json:"image_url"`
	Edition         string   `json:"edition"`
	Authors         []string `json:"authors"`
	Publisher       string   `json:"publisher"`
} //@name Book

type PaginatedBooks = util.PaginatedList[Book] //@name PaginatedBooks
